docs/source/libraries/sklearn.rst

```diff
@@ -209,6 +209,42 @@ long as:
   ``transform_feature_names`` for transformer classes not handled (yet) by ELI5
   or to override the default implementation.
 
+For instance, imagine a transformer which selects every second feature::
+
+    from sklearn.base import BaseEstimator, TransformerMixin
+    from sklearn.utils.validation import check_array
+    from eli5 import transform_feature_names
+
+    class OddTransformer(BaseEstimator, TransformerMixin):
+        def fit(self, X, y=None):
+            # we store n_features_ for the sake of transform_feature_names
+            # when in_names=None:
+            self.n_features_ = check_array(X).shape[1]
+            return self
+
+        def transform(self, X):
+            return check_array(X)[:, 1::2]
+
+    @transform_feature_names.register
+    def odd_feature_names(transformer, in_names=None):
+        if in_names is None:
+            from eli5.sklearn.utils import get_feature_names
+            # generate default feature names
+            in_names = get_feature_names(num_features=transformer.n_features_)
```

Comment:
It won't work as-is because currently get_feature_names requires `clf` argument; it seems that's fine to make it optional.